src/pocapi.hpp

```diff
@@ -14,5 +14,70 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 //*****************************************************************************
+#include <stddef.h>
+#include <stdint.h>
+
+struct OVMS_Status;
+
+// TODO reuse this in precision.hpp
+enum DataType {
+    OVMS_DATATYPE_BF16,
+    OVMS_DATATYPE_FP64,
+    OVMS_DATATYPE_FP32,
+    OVMS_DATATYPE_FP16,
+    OVMS_DATATYPE_I64,
+    OVMS_DATATYPE_I32,
+    OVMS_DATATYPE_I16,
+    OVMS_DATATYPE_I8,
+    OVMS_DATATYPE_I4,
+    OVMS_DATATYPE_U64,
+    OVMS_DATATYPE_U32,
+    OVMS_DATATYPE_U16,
+    OVMS_DATATYPE_U8,
+    OVMS_DATATYPE_U4,
+    OVMS_DATATYPE_U1,
+    OVMS_DATATYPE_BOOL,
+    OVMS_DATATYPE_CUSTOM,
+    OVMS_DATATYPE_UNDEFINED,
+    OVMS_DATATYPE_DYNAMIC,
+    OVMS_DATATYPE_MIXED,
+    OVMS_DATATYPE_Q78,
+    OVMS_DATATYPE_BIN
+};
+
+enum BufferType {
+    OVMS_BUFFERTYPE_CPU,
+    OVMS_BUFFERTYPE_CPU_PINNED,
+    OVMS_BUFFERTYPE_GPU,
+    OVMS_BUFFERTYPE_HDDL,
+};
+
+struct OVMS_Status;
+struct OVMS_InferenceRequest;
+struct OVMS_InferenceResponse;
+
+// OVMS_InferenceRequest
+OVMS_Status* OVMS_InferenceRequestNew(char* modelName, uint32_t servableVersion);
+OVMS_Status* OVMS_InferenceRequestDelete(OVMS_InferenceRequest* response);
+OVMS_Status* OVMS_InferenceRequestAddInput(OVMS_InferenceRequest* request, char* inputName, DataType datatype, uint64_t* shape, uint32_t dimCount);
```

Comment:
We could add shape struct and allow appending dimension to it:
```
OVMS_ShapeNew()
OVMS_ShapeDelete()
OVMS_ShapeAddDim(Shape*, int64_t)
```
This way we could get rid of shape ptr and dim count similarly to Single/Multi options in my PR